_posts/2019-08-15-release-1.9.0.md

```diff
@@ -0,0 +1,352 @@
+unify and power many forms of real-time and offline data processing
+applications as well as event-driven applications. In this release, we have
+made a huge step forward in that effort, by integrating Flink’s stream and
+batch processing capabilities under a single, unified runtime.
+
+Significant features on this path are batch-style recovery for batch jobs and
+a preview of the new Blink-based query engine for Table API and SQL queries.
+We are also excited to announce the availability of the State Processor API,
+which is one of the most frequently requested features and enables users to
+read and write savepoints with Flink DataSet jobs. Finally, Flink 1.9 includes
+a reworked WebUI and previews of Flink’s new Python Table API and its
+integration with the Apache Hive ecosystem.
+
+This blog post describes all major new features and improvements, important
+changes to be aware of and what to expect moving forward. For more details,
+check the [complete release
+changelog](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12315522&version=12344601).
+
+The binary distribution and source artifacts for this release are now
+available via the [Downloads](https://flink.apache.org/downloads.html) page of
+the Flink project, along with the updated
+[documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.9/).
+Flink 1.9 is API-compatible with previous 1.x releases for APIs annotated with
+the @Public annotation.
+
+Please feel encouraged to download the release and share your thoughts with
+the community through the Flink [mailing
+lists](https://flink.apache.org/community.html#mailing-lists) or
+[JIRA](https://issues.apache.org/jira/projects/FLINK/summary). As always,
+feedback is very much appreciated!
+
+
+{% toc %}
+
+
+## New Features and Improvements
+
+
+### Fine-grained Batch Recovery (FLIP-1)
+
+The time to recover a batch (DataSet, Table API and SQL) job from a task
+failure was significantly reduced. Until Flink 1.9, task failures in batch
+jobs were recovered by canceling all tasks and restarting the whole job, i.e,
+the job was started from scratch and all progress was voided. With this
+release, Flink persists intermediate results at network shuffle boundaries and
+use this data to limit the recovery to only those tasks that are affected by
+the failure. So-called failover regions of tasks which are connected via
+pipelined data exchanges define the boundaries among which tasks are affected
+by failures. More details are available in
+[FLIP-1](https://cwiki.apache.org/confluence/display/FLINK/FLIP-1+%3A+Fine+Grained+Recovery+from+Task+Failures).
+
+![alt_text]({{site.baseurl}}/img/blog/release-19-flip1.png "Fine-grained Batch
+Recovery")
+
+To use this new failover strategy, you need to do the following settings:
+
+ * Make sure you have the entry `jobmanager.execution.failover-strategy:
+   region` in your `flink-conf.yaml`.
+ * Set the `ExecutionMode` to `BATCH` in the `ExecutionConfig`.
```

Comment:
Setting the `ExecutionMode` is not needed if users want to use region failover strategy for “embarrassingly
parallel” streaming jobs. Setting it to `BATCH` does not break anything though.

Reply:
I'm not sure if I fully understand your comment.

This section is about the new fine-grained batch recovery. In my understanding, both settings need to be set to use it.
With the last paragraph (`The "Region" failover strategy also improves the recovery ... `) this is actually quite clear.

Reply:
Hi Robert, sorry for the confusion. I think your statement is Ok for users and we can leave it as is.

I just meant it can be more accurate technically.
Setting `jobmanager.execution.failover-strategy: region` is to enable the strategy, which recovers failed tasks in region granularity.
Setting `ExecutionMode` to `BATCH` is to break DataSet batch jobs into more regions. This improves the region failover experience but is not a necessary part to enable region failover. It does not work for batch SQL cases as well. However, it also does have negative effects for other cases(DataStream/SQL), so I think it's fine for users to always set it to be `BATCH`.

Reply:
Thanks for your comment @zhuzhurk, I think you are right. I makes sense to clarify the purpose of these parameters and how they interact.
@rmetzger, I'll suggest a change.